### ISSUE TITLE
Build failure on linux 5.10

### ISSUE THREAD

**Issue by szbergeron** [maaarghk/soundwire-dkms#4]
```
CC [M]  /var/lib/dkms/soundwire/1.4.0/build/usb/helper.o
/var/lib/dkms/soundwire/1.4.0/build/usb/endpoint.c: In function ‘queue_pending_output_urbs’:
/var/lib/dkms/soundwire/1.4.0/build/usb/endpoint.c:313:10: warning: parameter names (without types) in function declaration
  313 |   struct snd_usb_packet_info *uninitialized_var(packet);
      |          ^~~~~~~~~~~~~~~~~~~
/var/lib/dkms/soundwire/1.4.0/build/usb/endpoint.c:319:4: error: ‘packet’ undeclared (first use in this function)
  319 |    packet = ep->next_packet + ep->next_packet_read_pos;
      |    ^~~~~~
/var/lib/dkms/soundwire/1.4.0/build/usb/endpoint.c:319:4: note: each undeclared identifier is reported only once for each function it appears in
make[3]: *** [scripts/Makefile.build:279: /var/lib/dkms/soundwire/1.4.0/build/usb/endpoint.o] Error 1
make[3]: *** Waiting for unfinished jobs....
  LD [M]  /var/lib/dkms/soundwire/1.4.0/build/pci/hda/snd-hda-codec-ca0132.o
make[2]: *** [scripts/Makefile.build:496: /var/lib/dkms/soundwire/1.4.0/build/usb] Error 2
  LD [M]  /var/lib/dkms/soundwire/1.4.0/build/soc/snd-soc-core.o
make[1]: *** [Makefile:1805: /var/lib/dkms/soundwire/1.4.0/build] Error 2
make[1]: Leaving directory '/usr/lib/modules/5.10.0-1-mainline/build'
make: *** [Makefile:10: all] Error 2
```

This is using the latest-sofproject branch that was working with 5.8 and 5.9

**Comment by szbergeron** [comment]
Formatted as a screenshot:
![image](https://user-images.githubusercontent.com/7364398/102134801-05fb4f00-3e25-11eb-9038-9d0f4dbd62b5.png)

**Comment by maaarghk** [comment]
.... we not in kernel 5.10 without dkms?

**Comment by szbergeron** [comment]
Opening pacmixer I only see this:
![image](https://user-images.githubusercontent.com/7364398/102134877-1dd2d300-3e25-11eb-85b6-acfb14eb0660.png)

Doing a sound test with gnome settings I hear nothing

**Comment by szbergeron** [comment]
Have you seen reports of it working OOTB on 5.10?

Am currently using mainline from the arch "miffe" unofficial repo, should be 1:1 with arch linux-mainline

**Comment by maaarghk** [comment]
i haven't but i'd expect it cause the sku shows up in the kernel source... i'll try it myself just now and see what i can find out

**Comment by maaarghk** [comment]
5.10 should support it without issue, but CONFIG_SND_SOC_SOF_INTEL_SOUNDWIRE is not enabled in either manjaro or miffe's 5.10 config. I made a request to manjaro. Miffe does not maintain a custom config so I made a separate request to heftig for arch core config - no idea how quickly that will make its way back to miffe's thing as he said something i didn't quite understand about how he only merges that file every rc1? anyway, if my arch bug report is answered first then you could just wait for arch to release kernel 5.10

**Comment by maaarghk** [comment]
(someone else already made the request to debian)

**Comment by szbergeron** [comment]
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=976791 is giving me grand dreams about compiling my own kernel so I'll be back from that adventure either wiser or more frustrated than ever :P 

Do you by any chance have a link to the mentioned arch bug report?

E: nvm, tracked it down: https://bugs.archlinux.org/task/68978?project=1&order=dateopened&sort=desc

**Comment by szbergeron** [comment]
Have manually built mainline with the configs mentioned in that bug report but audio still doesn't seem to work for me. @maaarghk have you had any success?

**Comment by maaarghk** [comment]
hey, I applied this patch:

```patch
From 5d0a51a00e921e45d0efef931319db88496fc1e7 Mon Sep 17 00:00:00 2001
From: Mark Gallagher <mark@fts.scot>
Date: Wed, 16 Dec 2020 11:40:30 +0000
Subject: [PATCH] 5.10.1.arch2-1

Fix FS#68978 - Enable Soundwire drivers
---
 trunk/config | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/trunk/config b/trunk/config
index 12a5a81..5fe7ffd 100644
--- a/trunk/config
+++ b/trunk/config
@@ -1,6 +1,6 @@
 #
 # Automatically generated file; DO NOT EDIT.
-# Linux/x86 5.10.1-arch1 Kernel Configuration
+# Linux/x86 5.10.1-arch2 Kernel Configuration
 #
 CONFIG_CC_VERSION_TEXT="gcc (GCC) 10.2.0"
 CONFIG_CC_IS_GCC=y
@@ -7186,7 +7186,7 @@ CONFIG_SND_SOC_INTEL_SKYLAKE_SSP_CLK=m
 CONFIG_SND_SOC_INTEL_SKYLAKE_COMMON=m
 CONFIG_SND_SOC_ACPI_INTEL_MATCH=m
 CONFIG_SND_SOC_INTEL_MACH=y
-# CONFIG_SND_SOC_INTEL_USER_FRIENDLY_LONG_NAMES is not set
+CONFIG_SND_SOC_INTEL_USER_FRIENDLY_LONG_NAMES=y
 CONFIG_SND_SOC_INTEL_HASWELL_MACH=m
 CONFIG_SND_SOC_INTEL_BDW_RT5650_MACH=m
 CONFIG_SND_SOC_INTEL_BDW_RT5677_MACH=m
@@ -7223,6 +7223,7 @@ CONFIG_SND_SOC_INTEL_CML_LP_DA7219_MAX98357A_MACH=m
 CONFIG_SND_SOC_INTEL_SOF_CML_RT1011_RT5682_MACH=m
 CONFIG_SND_SOC_INTEL_SOF_DA7219_MAX98373_MACH=m
 CONFIG_SND_SOC_INTEL_EHL_RT5660_MACH=m
+CONFIG_SND_SOC_INTEL_SOUNDWIRE_SOF_MACH=m
 CONFIG_SND_SOC_MTK_BTCVSD=m
 CONFIG_SND_SOC_SOF_TOPLEVEL=y
 CONFIG_SND_SOC_SOF_PCI=m
@@ -7387,6 +7388,7 @@ CONFIG_SND_SOC_RT286=m
 CONFIG_SND_SOC_RT298=m
 CONFIG_SND_SOC_RT1011=m
 CONFIG_SND_SOC_RT1015=m
+CONFIG_SND_SOC_RT1308=m
 CONFIG_SND_SOC_RT1308_SDW=m
 CONFIG_SND_SOC_RT5514=m
 CONFIG_SND_SOC_RT5514_SPI=m
-- 
2.29.2

```

on top of https://github.com/archlinux/svntogit-packages/tree/packages/linux and sound works fine. i sent an e-mail with the patch to try and get it applied

**Comment by szbergeron** [comment]
Somehow that doesn't seem to be working for me, I'll try again later ig.

**Comment by maaarghk** [comment]
it is now on manjaro and arch repos (in testing on the latter), so i will consider this out of scope :)

**Comment by maaarghk** [comment]
@szbergeron Are you seeing anything useful in `journalctl -b`?

**Comment by szbergeron** [comment]
I haven't seen an update come down yet on the testing repos, not sure if it's been built.

Alsa does seem rather unhappy now though:
![image](https://user-images.githubusercontent.com/7364398/102388384-aa090580-3f9f-11eb-9d61-d51ed9ffd707.png)

**Comment by maaarghk** [comment]
what version of pulseaudio, alsa-lib, sof-firmware ?

**Comment by szbergeron** [comment]
![image](https://user-images.githubusercontent.com/7364398/102397681-36b9c080-3fac-11eb-9b5a-9011a3dae36d.png)

**Comment by maaarghk** [comment]
hmm,
```
mark@applecross ~ > pacman -Q --info alsa-lib | grep Version
Version         : 1.2.4-3
mark@applecross ~ > pacman -Q --info sof-firmware | grep Version
Version         : 1.6-3
mark@applecross ~ > pacman -Q --info alsa-ucm-conf | grep Version
Version         : 1.2.4-2
mark@applecross ~ > pacman -Q --info pulseaudio-alsa | grep Version
Version         : 1:1.2.2-2
mark@applecross ~ > pacman -Q --info pulseaudio | grep Version
Version         : 14.0-1
```

everything works fine here - I suppose just wait for the next kernel build to be done by arch and see what happens.
